Source/Core/UICommon/GameFile.cpp

```diff
@@ -624,23 +625,83 @@ std::string GameFile::GetNetPlayName(const Core::TitleDatabase& title_database)
   return name + " (" + ss.str() + ")";
 }
 
+static std::array<u8, 20> GetHash(u32 value)
+{
+  auto data = Common::BitCastToArray<u8>(value);
+  std::array<u8, 20> hash;
+  mbedtls_sha1_ret(reinterpret_cast<const unsigned char*>(data.data()), data.size(), hash.data());
+  return hash;
+}
+
+static std::array<u8, 20> GetHash(std::string_view str)
+{
+  std::array<u8, 20> hash;
+  mbedtls_sha1_ret(reinterpret_cast<const unsigned char*>(str.data()), str.size(), hash.data());
+  return hash;
+}
+
+static std::optional<std::array<u8, 20>> GetFileHash(const std::string& path)
+{
+  std::string buffer;
+  if (!File::ReadFileToString(path, buffer))
+    return std::nullopt;
+  return GetHash(buffer);
+}
+
+static std::optional<std::array<u8, 20>> MixHash(const std::optional<std::array<u8, 20>>& lhs,
```

Comment:
What do you think about making lhs and the return type non-optional? I feel like it would clean things up.

Reply:
You... think so? I feel like it would make `GameFile::GetSyncHash()` way more complicated, actually.

Reply:
Can't you initialize the hash to an all-zero vector, remove the `value_or` call at the end, and keep the rest as it is?

Reply:
https://gist.github.com/AdmiralCurtiss/7932ee2a8caf818f5e8559c8090d6b4b I guess? I don't really feel like either version is better than the other honestly.

Reply:
Ah, I forgot that `GetFileHash` returned an optional. I guess it's a wash then. Feel free to ignore my suggestion.